gix-pack/src/data/file/decode/entry.rs

```diff
@@ -177,13 +177,9 @@ impl File {
         use crate::data::entry::Header::*;
         match entry.header {
             Tree | Blob | Commit | Tag => {
-                out.resize(
-                    entry
-                        .decompressed_size
-                        .try_into()
-                        .expect("size representable by machine"),
-                    0,
-                );
+                let size = entry.decompressed_size.try_into().map_err(|_| Error::OutOfMemory)?;
+                out.try_reserve_exact(size).map_err(|_| Error::OutOfMemory)?;
```

Comment:
It's easy to fall into the trap of interpreting [`additional` items to reserve](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.try_reserve) as the total target size. This code would attempt to reserve more memory than it needs.